### DISCUSSION TITLE
Effect of `del` on Unused Parameters in Jitted JAX Functions

### DISCUSSION THREAD

**Discussion by helpingstar** [jax-ml/jax#23161] [Q&A] [answered]
Hello,

Thank you very much for always providing kind explanations to many questions. While studying the `optax` code, I noticed the following pattern: when a function is designed to have consistent parameter signatures for practical purposes, unused parameters are immediately deleted using `del`, assuming that these parameters are obviously not needed.

In the example below, it is `del params`.

Does this provide any actual benefit in terms of memory usage? I thought that when a function is jitted, unused parameters might be optimized away or the `del` statement might be ignored (similar to how `print` is handled).

However, seeing this code pattern used throughout, I am curious whether this actually offers advantages in terms of speed or memory efficiency.

![image](https://github.com/user-attachments/assets/9242d8f7-dd43-452c-bc5a-311d00f41d7a)

https://github.com/google-deepmind/optax/blob/51b5d419512423408f31a4d8c10803f51fe9e12b/optax/_src/base.py#L223-L240

![image](https://github.com/user-attachments/assets/03f56f30-0ef5-4e5a-b3e0-895ca44419c5)

https://github.com/google-deepmind/optax/blob/51b5d419512423408f31a4d8c10803f51fe9e12b/optax/_src/base.py#L263-L265

**Comment by dfm** [comment, marked as answer]
I'm not sure this is the right place to open an optax discussion :D, but...

No, this doesn't provide any performance benefits! This pattern is adopted from [Google's Python style guide](https://google.github.io/styleguide/pyguide.html#21-lint), and it can be useful for making sure that you actually don't use variables, and it can appease the little red squiggles in some editors.

**Reply by helpingstar** [reply]
Thank you @dfm for the clarification! I appreciate your detailed response. I’ll make sure to ask questions in the appropriate channels in the future. Thanks again for your help!
